test.js

```diff
@@ -283,6 +289,64 @@ test('responseTime for request emitting error event', function (t) {
   })
 })
 
+if (process.stdout.writableEnded === undefined) {
+  // CI: This only runs on Node 10
+  test("throw error if 'onlyLogFinishedRequests': false and Node.js doesn't have writableEnded", function (t) {
+    var dest = split(JSON.parse)
+    var throwFunction = function () {
+      pinoHttp({
+        onlyLogFinishedRequest: false
+      }, dest)
+    }
+    t.throws(throwFunction, { message: "'onlyLogFinishedRequest': false requires Node v12.9.0 or later" })
+    t.end()
+  })
+} else {
+  // CI: Do not run this test on versions of Node that don't support it
+  // https://github.com/pinojs/pino-http/issues/116
+  test('finished for simple request', function (t) {
+    var dest = split(JSON.parse)
+    var logger = pinoHttp({
+      onlyLogFinishedRequest: true
+    }, dest)
+
+    setup(t, logger, function (err, server) {
+      t.error(err)
+      var request = doGet(server, WAIT_URL)
+      request.on('error', function (error) {
+        t.equal(error.message, 'socket hang up')
+        setTimeout(() => {
+          var line = dest.read()
+          t.equal(line, null)
+          t.end()
+        }, 100)
```

Comment:
Can you use a `setImmediate()` instead?

Reply:
Hm... The tests were the only thing I didn't rewrite; I can take a look at these later today. I don't see why not.

Reply:
@mcollina: I wasn't able to completely replace `setTimeout` with `setImmediate` without a triple nested `setImmediate` which seemed worse from a clarity standpoint. The author's original intent was to give the request enough time to start so a failure could be simulated and ultimately logged. One or two of these `setTimeout` can be easily replaced with `setImmediate`. To completely remove `setTimeout` would require more work than I can set aside right now. I may be able to revisit later -- or anyone else is free to pick this up as well! It's a good task for those looking to learn about the particulars of the event loop.

Reply:
using setTimeout will lead to flaky CI runs, I really need this fixed before landing.

Another approach is using some logical events instead of waiting.